python/semantic_kernel/agents/open_ai/responses_agent_thread_actions.py

```diff
@@ -434,10 +462,45 @@ async def invoke_stream(
                                 choice_index=request_index,
                             )
                             all_messages.append(msg)
+                        case ResponseMcpCallArgumentsDoneEvent():
+                            if on_intermediate_message:
+                                # Get tool information from stored state
+                                tool_info: dict[str, Any] = mcp_tool_calls.get(event.item_id, {})
+                                tool_name: str = tool_info.get("name", "")
+                                server_label: str = tool_info.get("server_label", "")
+
+                                mcp_args_done = FunctionCallContent(
```

Comment:
We'd need to add similar handling for the non-streaming case, correct?

Reply:
@moonbox3 I tried making changes.
https://github.com/microsoft/semantic-kernel/pull/13112/commits/5c683c584e6876d542c566a7897443915c55b26a